### PR TITLE
feat: adds time range module

### PR DESCRIPTION
**Pull request by linnall**
# Pull Request Checklist

- [ ] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [ ] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [ ] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [ ] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [ ] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

## PR-Codex overview
This PR introduces a new `TimeRangeModule` for managing time-based operations in the smart contract system. It adds functionality for encoding installation and uninstallation data based on time constraints and includes tests for these functionalities.

### Detailed summary
- Commented out `EntityIdOverrideError` and `DEFAULT_OWNER_ENTITY_ID` imports in `installValidation.ts`.
- Added `TimeRangeModule` with methods for encoding install and uninstall data in `module.ts`.
- Implemented tests for installing and uninstalling the `TimeRangeModule` in `client.test.ts`, verifying both valid and invalid time ranges.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: account-kit/smart-contracts/src/ma-v2/modules/time-range-module/module.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import { encodeAbiParameters, type Address, type Hex } from "viem";
+
+import { timeRangeModuleAbi } from "./abis/timeRangeModuleAbi.js";
+
+const addresses = {
+  default: "0x00", // TO DO: replace with deployed module address?
```
Comment (dphilipson) [inline account-kit/smart-contracts/src/ma-v2/modules/time-range-module/module.ts:6]: As mentioned in other PR, use `zeroAddress` from Viem.

File: account-kit/smart-contracts/src/ma-v2/modules/time-range-module/module.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+} as Record<number | "default", Address>;
+
+const meta = {
+  name: "TimeRangeModule",
+  version: "alpha.1",
+  addresses,
+};
+
+// Todo: some unified type for ERC-6900 v0.8 modules. I couldn't figure out how to parameterize the class itself over the abi type parameters for onInstall and onUninstall.
```
Comment (dphilipson) [inline account-kit/smart-contracts/src/ma-v2/modules/time-range-module/module.ts:15]: What might work:
```ts
type Module<TAbi extends Abi, InstallData, UninstallData> = {
  meta: ModuleMeta,
  abi: TAbi,
  encodeOnInstallData: (args: InstallData) => Hex;
  encodeOnUninstallData: (args: UninstallData) => Hex;
};
```
Let me know if you want to jam on this.

File: account-kit/smart-contracts/src/ma-v2/modules/time-range-module/module.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    validUntil: number;
+    validAfter: number;
+  }): Hex => {
+    const { entityId, validUntil, validAfter } = args;
+
+    return encodeAbiParameters(
+      [
+        {
+          type: "uint32",
+          value: entityId,
```
Comment (dphilipson) [inline account-kit/smart-contracts/src/ma-v2/modules/time-range-module/module.ts:30]: As in other PR: the `value` field doesn't do anything here, should be just `type` and optionally `name`.

**Review by dphilipson — APPROVED** [review]

File: account-kit/smart-contracts/src/ma-v2/modules/time-range-module/module.ts (hunk 1 above)
Comment (linnall) [reply account-kit/smart-contracts/src/ma-v2/modules/time-range-module/module.ts:6]: `addresses` is no longer part of the module definition, therefore resolving this comment!

File: account-kit/smart-contracts/src/ma-v2/modules/time-range-module/module.ts (hunk 2 above)
Comment (linnall) [reply account-kit/smart-contracts/src/ma-v2/modules/time-range-module/module.ts:5]: created an [Asana ticket](https://app.asana.com/0/1205598840815267/1209002631957336/f) to address this for the next iteration of the MAv2 SDK!

**Pull request merged**